src/avatar_action.cpp

```diff
@@ -1006,16 +1009,16 @@ void avatar_action::plthrow( avatar &you, item_location loc,
         }
     }
     // if you're wearing the item you need to be able to take it off
-    if( you.is_wearing( thrown.typeId() ) ) {
-        ret_val<bool> ret = you.can_takeoff( thrown );
+    if( you.is_wearing( orig->typeId() ) ) {
+        ret_val<bool> ret = you.can_takeoff( *orig );
         if( !ret.success() ) {
             add_msg( m_info, "%s", ret.c_str() );
             return;
         }
     }
     // you must wield the item to throw it
-    if( !you.is_wielding( thrown ) ) {
-        you.wield( thrown );
+    if( !you.is_wielding( *orig ) ) {
+        you.wield( *orig );
```

Comment:
This removes the item from the main inventory / the worn item list. The item pointer `orig` will be invalid after that,

Reply:
Dosnt seem to cause any problems.

After it's wielded, further operations are called on it, like if( orig->count_by_charges() ) etc, and its fine. 
Though I suppose that could be changed to `you.weapon`  ?

Reply:
Well, ive redefined `orig` as `you.weapon` after wielding, added protection for failed wielding to return if wielding prompt is escaped out of, and continued to use you.weapon afterwards anyway , just incase pointer is invalidated.